src/Numerical/RootFinding/QuadraticEquation.cpp

```diff
@@ -0,0 +1,24 @@
+
+#include "Numerical/RootFinding/QuadraticEquation.hpp"
+
+#include <gsl/gsl_poly.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/ExpectsAndEnsures.hpp"
+
+double positive_root(const double a, const double b, const double c) {
+  double x0;
+  double x1;
+  gsl_poly_solve_quadratic(a, b, c, &x0, &x1);
+  Ensures(x0 <= 0.0 and x1 >= 0.0);
+  return x1;
```

Comment:
Maybe you were being loose in your example, but note that the function is unsafe when `gsl_poly_solve_quadratic` returns 0 or 1, and is safe when it returns 2.